tests/cpgrid/addLgrsOnDistributedGrid_test.cpp

```diff
@@ -753,6 +876,57 @@ BOOST_AUTO_TEST_CASE(distributed_in_all_ranks_lgr)
         // 64 new refined cells. 5x5x5 = 125 points (only 98 = 125 - 3x3x3 parent corners new points - new global ids).
         grid.addLgrsUpdateLeafView(cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
         refinePatch_and_check(grid, cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
+
+        // Check global id is not duplicated for points for each LGR
+        // LGR1 dim 4x4x4 -> 5x5x5 = 125 points
+        std::vector<int> local_point_ids;
+        local_point_ids.reserve(125); // expected_point_ids in LGR1
+        for (const auto& element : elements(grid.levelGridView(1))) {
+            for (int corner = 0; corner < 8; ++corner)
+            {
+                const auto& point = element.subEntity<3>(corner);
+                local_point_ids.push_back(grid.currentData()[1]->globalIdSet().id(point));
+            }
+        }
+        auto [all_point_ids, displPoint ] = Opm::allGatherv(local_point_ids, grid.comm());
+        const std::set<int> all_point_ids_set(all_point_ids.begin(), all_point_ids.end());
+        // Difference with preveious test case:
+        // Global cell ids of cells to be refined = {1,2,5,6,13,14,17,18}
+        // cell 1 = { interior in P0, overlap in P1, overlap in P2, does not exist in P3}
+        // cell 2 = { overlap in P0, does not exist in P1, interior in P2, overlap in P3}
+        // cell 5 = { interior in P0, overlap in P1, overlap in P2, does not exist in P3}
+        // cell 6 = { overlap in P0, does not exist in P1, interior in P2, does not exist in P3}
+        // cell 13 = { overlap in P0, interior in P1, does not exist in P2, overlap in P3}
+        // cell 14 = { does not exist in P0, overlap in P1, overlap in P2, interior in P3}
+        // cell 17 = { overlap in P0, interior in P1, overlap in P2, does not exist in P3}
+        // cell 18 = { does not exist in P0, overlap in P1, interior in P2, overlap in P3}
+        // Remark: the LGR is distributed in ALL processes BUT
+        // - P0 does NOT see cell 18, but sees all the others.
+        // - P1 does NOT see cells 2 and 6, but sees the rest of them.
+        // - P2 does NOT see cell 13, but sees all the others.
+        // - P3 does NOT see cell 1, 5 and 17, but sees all the others.
+        // Remark: the LGR is distributed only in 2 processes which are the only two processes seeing these cells.
```

Comment:
I don't understand this sentence. Are there more than 1 LGR (which is said to be distributed among all processes above)

Reply:
I will try to improve the comment. There is only one LGR here. The block of parent cells  has dimension 2x2x2, the parent cell indices are {1,2,5,6,13,14,17,18}. Each parent cell gets refined into 2x2x2 children, therefore the LGR dimension is 4x4x4.